eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -69,23 +81,131 @@ object Databases extends Logging {
     override val network = new SqliteNetworkDb(networkJdbc)
     override val audit = new SqliteAuditDb(auditJdbc)
     override val channels = new SqliteChannelsDb(eclairJdbc)
     override val peers = new SqlitePeersDb(eclairJdbc)
     override val payments = new SqlitePaymentsDb(eclairJdbc)
     override val pendingRelay = new SqlitePendingRelayDb(eclairJdbc)
+    override def backup(backupFile: File): Unit = {
+      val tmpFile = new File(backupFile.getAbsolutePath.concat(".tmp"))
 
-    override def backup(file: File): Unit = {
       SqliteUtils.using(eclairJdbc.createStatement()) {
-        statement => {
-          statement.executeUpdate(s"backup to ${file.getAbsolutePath}")
+        statement =>  {
+          statement.executeUpdate(s"backup to ${tmpFile.getAbsolutePath}")
         }
       }
+
+      // this will throw an exception if it fails, which is possible if the backup file is not on the same filesystem
+      // as the temporary file
+      Files.move(tmpFile.toPath, backupFile.toPath, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE)
     }
+    override val isBackupSupported: Boolean = true
+
+    override def obtainExclusiveLock(): Unit = ()
   }
+
+  def setupPsqlDatabases(dbConfig: Config, datadir: File, lockExceptionHandler: LockExceptionHandler): Databases = {
+    val database = dbConfig.getString("psql.database")
+    val host = dbConfig.getString("psql.host")
+    val port = dbConfig.getInt("psql.port")
+    val username = if (dbConfig.getIsNull("psql.username") || dbConfig.getString("psql.username").isEmpty)
+      None
+    else
+      Some(dbConfig.getString("psql.username"))
+    val password = if (dbConfig.getIsNull("psql.password") || dbConfig.getString("psql.password").isEmpty)
+      None
+    else
+      Some(dbConfig.getString("psql.password"))
+    val properties = {
+      val poolConfig = dbConfig.getConfig("psql.pool")
+      Map.empty
+        .updated("max-size", poolConfig.getInt("max-size").toLong)
+        .updated("connection-timeout", poolConfig.getDuration("connection-timeout").toMillis)
+        .updated("idle-timeout", poolConfig.getDuration("idle-timeout").toMillis)
+        .updated("max-life-time", poolConfig.getDuration("max-life-time").toMillis)
+
+    }
+    val lockType = LockType(dbConfig.getString("psql.lock-type"))
+    val leaseInterval = dbConfig.getDuration("psql.ownership-lease.lease-interval").toSeconds.seconds
+    val lockTimeout = dbConfig.getDuration("psql.ownership-lease.lock-timeout").toSeconds.seconds
+
+    Databases.postgresJDBC(
+      database = database, host = host, port = port,
+      username = username, password = password,
+      poolProperties = properties,
+      databaseLeaseInterval = leaseInterval, lockTimeout = lockTimeout,
+      lockExceptionHandler = lockExceptionHandler, lockType = lockType, datadir = datadir
+    )
+  }
+
+  private def checkIfDatabaseUrlIsUnchanged(url: String, datadir: File ): Unit = {
```

Comment:
So this is a safeguard agains pointing the node to a wrong database and losing the funds. In case of SQLite the seed and the database are stored in the same place so it's not an issue.

So this method prevents node from starting if the database url has been changed accidentally. If the node operator can remove `last_jdbcurl` file to force using the new database.

Probably it's not the best solution. As an alternative we can store node ID inside Postgres database and prevent the node from staring if it's public key does not match the node id in the database.